Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -208,13 +208,13 @@ def adaptationNoteLinter : TextbasedLinter := fun lines ↦ Id.run do
 /-- Lint a collection of input strings if one of them contains trailing whitespace. -/
 def trailingWhitespaceLinter : TextbasedLinter := fun lines ↦ Id.run do
   let mut errors := Array.mkEmpty 0
-  let mut fixedLines := lines
+  let mut fixedLines : Vector String lines.size := lines.toVector
```

Comment:
Can you say why you need this conversion of an array to a vector and back? (And need I worry about that allocating - if not, I'm happy to let this slide as-is.) I know this code is not performance-sensitive, but if there's an "obviously better" way that's equally hard, I'd like to choose that.

Reply:
Yes, the allocation is fine:
```
set_option trace.compiler.ir.result true in
def foo (x : Array Nat) := 
  x.toVector.size
```
prints
```
[result]
def Mathlib.Meta.FunProp.foo (x_1 : @& obj) : obj :=
  let x_2 : obj := Array.size ◾ x_1;
  ret x_2
def Mathlib.Meta.FunProp.foo._boxed (x_1 : obj) : obj :=
  let x_2 : obj := Mathlib.Meta.FunProp.foo x_1;
  dec x_1;
  ret x_2
```
(ie it disappears by the time we get to compiled code).